### PR TITLE
Add Python Binding

### PR DESCRIPTION
**Pull request by TopKeyboard**
- Add Python Binding
- Refactor build scripts

### CODE REVIEW

File: engine/db/wal/write_ahead_log.hpp [hunk 1]
```diff
@@ -58,7 +58,7 @@ class WriteAheadLog {
       RotateFile();
     }
     int64_t next = global_counter_.IncrementAndGet();
-    fprintf(file_, "%lld %d %s\n", next, type, entry.c_str());
+    fprintf(file_, "%ld %d %s\n", next, type, entry.c_str());
```
Comment (richard-epsilla) [inline engine/db/wal/write_ahead_log.hpp:61]: This part is interesting. lld will not warning on macos. ld will not warning on linux. Cannot satisfy both

File: engine/cmd/main.cpp
Comment (richard-epsilla) [inline engine/cmd/main.cpp:1]: Need fetch latest code to resolve conflicts

File: engine/bindings/python/interface.h [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+#include <memory>
+#include "db/db_server.hpp"
+static PyObject *EpsillaError;
+
+static PyObject *spam_system(PyObject *self, PyObject *args);
+static PyObject *load_db(PyObject *self, PyObject *args);
+static PyObject *use_db(PyObject *self, PyObject *args);
+static PyObject *create_table(PyObject *self, PyObject *args);
+static PyObject *insert(PyObject *self, PyObject *args);
+static PyObject *query(PyObject *self, PyObject *args);
```
Comment (richard-epsilla) [inline engine/bindings/python/interface.h:13]: There is also drop_table and unload_db

File: engine/bindings/python/interface.h [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+static std::string db_name;
+static vectordb::engine::DBServer *db;
+
+static PyMethodDef EpsillaMethods[] = {
+    {"system", spam_system, METH_VARARGS, "Execute a shell command."},
+    {"load_db", load_db, METH_VARARGS, "Load the database"},
+    {"use_db", use_db, METH_VARARGS, "Use the database"},
```
Comment (richard-epsilla) [inline engine/bindings/python/interface.h:21]: Why not all functions above exposed here?

Comment (TopKeyboard) [reply engine/bindings/python/interface.h:19]: this is draft PR.

File: engine/cmd/main.cpp
Comment (TopKeyboard) [reply engine/cmd/main.cpp:1]: will do have a major changes are done.

File: engine/bindings/python/interface.cpp [hunk 4]
```diff
@@ -68,6 +55,101 @@ static PyObject *use_db(PyObject *self, PyObject *args)
   db_name = namePtr;
   return PyLong_FromLong(0);
 }
-static PyObject *create_table(PyObject *self, PyObject *args) {}
+static PyObject *create_table(PyObject *self, PyObject *args)
```
Comment (richard-epsilla) [inline engine/bindings/python/interface.cpp:58]: does this mean we will need reimplement the interface and keep the native C++ and python binding implementation in sync from now on?

Comment (TopKeyboard) [reply engine/bindings/python/interface.cpp:58]: what do you mean by "in sync"? as long as all required arguments are passed to the python interface, the C++ interface can be different.

File: engine/bindings/python/interface.h (hunk 2 above)
Comment (TopKeyboard) [reply engine/bindings/python/interface.h]: added

File: engine/cmd/main.cpp
Comment (TopKeyboard) [reply engine/cmd/main.cpp:1]: resolved

File: engine/bindings/python/interface.h (hunk 3 above)
Comment (TopKeyboard) [reply engine/bindings/python/interface.h]: PR is competed and everything is added

File: engine/db/wal/write_ahead_log.hpp (hunk 1 above)
Comment (TopKeyboard) [reply engine/db/wal/write_ahead_log.hpp:66]: maybe should use `%I64d`

**Review by richard-epsilla — APPROVED** [review]

**Pull request merged**